src/Strategy/Maven/Plugin.hs

```diff
@@ -151,11 +152,11 @@ textArtifactToPluginOutput :: Has Diagnostics sig m => Tree TextArtifact -> m Pl
 textArtifactToPluginOutput
   ta = buildPluginOutput ta
     where
-      artifactNames :: [Text]
-      artifactNames = foldl' (\a c -> (artifactText c : a)) mempty ta
+      artifacts :: [TextArtifact]
+      artifacts = nub $ foldl' (flip (:)) mempty ta
```

Comment:
It looks like this is the only consequential change- can you explain why this fixes the issue? It is confusing to me that de-duplicating the list results in this issue being fixed.

Reply:
Cause of the defect, was due to the fact that we were indexing by the name, which was wrong since two artifact can have same name, but different (scope/classifier) and they should be treated differently in indexing. We nub the results to ensure two exact same TextArtifact have same numeric ID.